spec/src/main/asciidoc/background.asciidoc

```diff
@@ -0,0 +1,106 @@
+
+* Extract security token from the request
+  ** For RESTful services, this is usually achieved by obtaining the token from the Authorization header.
+* Perform validation checks against the token
+  ** This step usually depends on the token format and security protocol in use. The objective is make sure the token is valid and can be consumed by the application. It may involve signature, encryption and expiration checks.
+* Introspect the token and extract information about the subject
+  ** This step usually depends on the token format and security protocol in use. The objective is to obtain all the necessary information about the subject from the token.
+* Create a security context for the subject
+  ** Based on the information extracted from the token, the application creates a security context for the subject in order to use the information wherever necessary when serving protected resources.
+
+## Using JWT Bearer Tokens to Protect Services
+For now, use cases are based on a scenario where services belong to the same security domain. This is an important note in order to avoid dealing with all complexities when you need to access services across different security domains. With that in mind, we assume that any information carried along with a token could be understood and processed (without any security breaches) by the different services involved.
+
+The use case can be described as follows:
+
+. Client sends a HTTP request to Service A including the JWT as a bearer token:
+----
+    GET /resource/1 HTTP/1.1
+    Host: example.com
+    Authorization: Bearer mF_9.B5f-4.1JqM
+----
+. Token-Based Authentication Mechanism in front of Service A perform all steps described on the Token-Based Authentication Section.
+. Token-Based Authentication Mechanism perform a role and group mapping based on the JWT claims. Where the role and group mapping is fully configurable by the application.
+
+https://tools.ietf.org/html/rfc7519[JWT] tokens follow a well defined and known standard that is becoming the most common token format to protect services. It not only provides a token format but additional security aspects like signature and encryption based on another set of standards like JWS, JWE and others.
+
+There are few reasons why JWT is becoming so widely adopted:
+
+* Token validation doesn’t require an additional trip and can be validated locally by each service
+* Given its JSON nature, it is solely based on claims or attributes to carry authentication and authorization information about a subject.
+* Makes easier to support different types of access control mechanisms such as ABAC, RBAC, Context-Based Access Control, etc.
+* Message-level security using signature and encryption as defined by both JWS and JWE standards
+* Given its JSON nature, processing JWT tokens becomes trivial and lightweight. Especially if considering JEE standards such as JSON-P or the different third-party libraries out there such as Nimbus, Jackson, etc.
```

Comment:
I know this is being picky, but could we modify JEE to Java EE?  This provides a nice counterpart to the Java SE nomenclature and is more industry accepting.  Thanks.  I've noticed this a couple of times in this write-up.  When you make this change, please do a "search all" to find all occurrences.  Thanks.